Dockerfile

```
@@ -0,0 +1,63 @@
+# Stage 1: Build the binary
+FROM rust:1.89-slim AS builder
+
+WORKDIR /build
+
+# Install build dependencies (pkg-config needed for some crates)
+RUN apt-get update && apt-get install -y --no-install-recommends \
```

Hadolint (2.14.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)